src/classes/UTIL_CurrencyConversion.cls

```diff
@@ -0,0 +1,131 @@
+
+        return effRate;
+    }
+
+    /**
+     * @description Returns the converted amount to the toCurr using the proper exchange rate.
+     * @param amt - The amount to convert
+     * @param dt - Effective Date to use for conversion (or null to use the current date)
+     * @param fromCurr - The currency to convert from
+     * @param toCurr - The currency to convert to.
+     * @return Amount converted to the toCurr using the applied effective date
+     */
+    public static Decimal convertAmount(Decimal amt, Date dt, String fromCurr, String toCurr) {
+        if (!UTIL_Currency.getInstance().isMultiCurrencyOrganization() || fromCurr == toCurr) {
+            return amt;
+        } else {
+            Decimal effRate = getExchangeRate(dt, fromCurr, toCurr);
+            Integer decimalPlaces = UTIL_CurrencyCache.getInstance().getCurrDecimalPlaces(toCurr);
+            Decimal convertedToCorp = amt.divide(effRate, RATE_DECIMAL_PLACES, System.RoundingMode.HALF_UP) * 100;
+            Decimal convertedToTarget = Decimal.ValueOf(convertedToCorp.Round( System.RoundingMode.HALF_UP));
+            return convertedToTarget.divide(100, decimalPlaces, System.RoundingMode.HALF_UP);
+        }
+    }
+
+    /*******************************************************************************************************
+    * @description Converts an amount from the corporate currency to a given currency always using the
+    *   non-dated exchange rate exchange rate even if ACM is enabled in the org.
+    * @param ISO A string of the currency's iso code that we wish to convert the amount to.
+    * @param amt A decimal of the amount to convert.
+    * @return decimal The converted amount.
+    ********************************************************************************************************/
+    //do a conversion from amount in corporate currency to the currency specified by the ISO code
+    public static decimal convertFromCorporateUsingStandardRates(String toISO, Decimal amt) {
+        return convertAmountUsingStandardRates(UTIL_Currency.getInstance().getOrgDefaultCurrency(), toISO, amt);
+    }
+
+    /*******************************************************************************************************
+    * @description Converts an amount from any given currency to any other currency always using the
+    *   non-dated exchange rate exchange rate even if ACM is enabled in the org.
+    * @param fromISO A string of the currency's iso code of the amount we wish to convert.
+    * @param toISO A string of the currency's iso code that we wish to convert the amount to.
+    * @param amt A decimal of the amount to convert.
+    * @return decimal The converted amount.
+    ********************************************************************************************************/
+    public static decimal convertAmountUsingStandardRates(String fromISO, String toISO, Decimal amt) {
+        alwaysUseStandardCurrencyRate = true;
```

Comment:
What if the same transaction wants standard rates for one conversion, then dated for another conversion? For example, converting an amount on a non-ACM supported object, then for an Opportunity? Wouldn't this set all other conversions in that transaction to use standard rates?